[13.0][MIG] l10n_es_ticketbai_api

ao-landoo opened the pull request:
Migración l10n_es_ticketbai_api a V13.

Apenas tiene cambios de la versión anterior.

ljsalvatierra-binovo committed:
[ADD] l10n_es_ticketbai_api: send invoices to the Basque Country tax agencies (API).
Base addon for l10n_es_ticketbai.
Added Python library dependencies to requirements.txt.

ao-landoo committed:
[MIG] l10n_es_ticketbai_api: Migration to 12.0

oca-travis committed:
[UPD] Update l10n_es_ticketbai_api.pot

ao-landoo commented:
@pedrobaeza Después de ejecutar pre-commit y añadir los cambios de la migración a la 13 me está fallando con varios errores. Donde debería solucionar estos, ¿dentro del commit Migration to 13.0?

pedrobaeza commented:
Sí, los cambios de pre-commit que te pide ahora se deben a lo que tú hayas hecho de adaptación a la v13, por lo que eso debe ser parte del mismo commit.

ao-landoo commented:
Estoy obteniendo tres errores que no consigo quitar. Aunque no se pase de los 80 caracteres me está dando el error y ademas me aconseja dejarlo como lo tenía antes (que también me da error de la longitud de linea).

`l10n_es_ticketbai_api/models/ticketbai_invoice.py:680:94: B950 line too long (93 > 80 characters)
l10n_es_ticketbai_api/models/ticketbai_invoice.py:696:98: B950 line too long (97 > 80 characters)
l10n_es_ticketbai_api/models/ticketbai_invoice.py:702:94: B950 line too long (93 > 80 characters)`

pedrobaeza commented:
Eso es porque lo pones a 80 caracteres antes de pre-commit, pero el pre-commit siempre te lo cambiará (debes instalarlo y pasarlo en local para evitar sorpresas aquí). Tienes que acortar los nombres de variables, o almacenar cosas en variables intermedias para acortarlas, que por cierto `TicketBaiInvoiceResponseCode` no luce como un nombre de variable adecuado por muchas razones:

- No utiliza nomenclatura snake_case.
- Nombre largo que no añade nada. En ese contexto, con `response_code` o `invoice_response_code` debería ser suficiente. Ya se sabe que es de ticket BAI.

ao-landoo commented:
> Eso es porque lo pones a 80 caracteres antes de pre-commit, pero el pre-commit siempre te lo cambiará (debes instalarlo y pasarlo en local para evitar sorpresas aquí). Tienes que acortar los nombres de variables, o almacenar cosas en variables intermedias para acortarlas, que por cierto `TicketBaiInvoiceResponseCode` no luce como un nombre de variable adecuado por muchas razones:
> 
> * No utiliza nomenclatura snake_case.
> * Nombre largo que no añade nada. En ese contexto, con `response_code` o `invoice_response_code` debería ser suficiente. Ya se sabe que es de ticket BAI.

Me apunto lo de ejecutar el pre_commit en local

Me queda un error (o eso creo) que no se por donde cogerlo.

`l10n_es_ticketbai_api/ticketbai/api.py:17:1: B903 Data class should either be immutable or use __slots__ to save memory. Use collections.namedtuple to generate an immutable class, or enumerate the attributes in a __slot__ declaration in the class to leave attributes mutable.`

rp-landoo reviewed (approved):
Aprobación Funcional

ao-landoo committed:
[FIX] fix l10n_es_ticketbai_api module warnings

oca-travis committed:
[UPD] Update l10n_es_ticketbai_api.pot

oca-transbot committed:
Update translation files

Updated by "Update PO files to match POT (msgmerge)" hook in Weblate.

Translation: l10n-spain-12.0/l10n-spain-12.0-l10n_es_ticketbai_api
Translate-URL: https://translation.odoo-community.org/projects/l10n-spain-12-0/l10n-spain-12-0-l10n_es_ticketbai_api/

ao-landoo committed:
[MIG] l10n_es_ticketbai_api: Migration to 13.0

xAdrianC-Kernet commented:
Instalado, probado a alimentar instalaciones, certificados y añadir a compañía.

Todo Ok.

ao-landoo commented:
> Instalado, probado a alimentar instalaciones, certificados y añadir a compañía.
> 
> Todo Ok.

Gracias Adrian por las pruebas. Si todo ha funcionado bien como dices podrías realizar un 'approval' para poder mergear ya este módulo?

xAdrianC-Kernet reviewed (approved):
Instalado, probado a alimentar instalaciones, certificados y añadir a compañía.

Todo Ok.

xAdrianC-Kernet commented:
> > Instalado, probado a alimentar instalaciones, certificados y añadir a compañía.
> > Todo Ok.
> 
> Gracias Adrian por las pruebas. Si todo ha funcionado bien como dices podrías realizar un 'approval' para poder mergear ya este módulo?

Creo que así está hecho correcto, perdona que no llevo mucho participando por aqui... gracias por el trabajo

ao-landoo commented:
@pedrobaeza Se puede mergear este PR?

pedrobaeza commented on l10n_es_ticketbai_api/__manifest__.py:
```diff
@@ -4,7 +4,7 @@
 {
     "name": "TicketBAI - API",
-    "version": "12.0.0.1.0",
+    "version": "13.0.0.1.0",
```
Debería ser 13.0.1.0.0.

ao-landoo replied:
Arreglado

ao-landoo committed:
[FIX] l10n_es_ticketbai_api: state field label, l10n_es_ticketbai: installation wizard info

oca-travis committed:
[UPD] Update l10n_es_ticketbai_api.pot

OCA-git-bot committed:
l10n_es_ticketbai_api 12.0.0.1.1

ao-landoo committed:
[IMP] l10n_es_ticketbai_api: black, isort, prettier

ao-landoo commented:
@pedrobaeza Si lo ves correcto realiza merge para poder ponerme con el PR de l10n_es_ticketbai. Gracias!

pedrobaeza commented:
Bueno, mi revisión es de forma, no de fondo, pero tengo entendido que se quiere cambiar el sistema de asociación a los impuestos que no sea por descripción/nombre. ¿No es mejor esperar a ese cambio?

ao-landoo commented:
@pedrobaeza Ese cambio es en el módulo l10n_es_ticketbai, no influye para este módulo

pedrobaeza commented:
OK, fusionando en base a reviews existentes:

/ocabot merge nobump

pedrobaeza commented:
¿Podéis por favor hacer un nuevo PR para resolver los warnings generados en el runbot?

```
Two fields (test_qr_base_url, qr_base_url) of tbai.tax.agency() have the same label: QR Base URL.
Two fields (tax_agency_id, tbai_tax_agency_id) of res.company() have the same label: Tax Agency.
```
